libpod/runtime_ctr.go

```diff
@@ -687,11 +687,17 @@ func (r *Runtime) removeContainer(ctx context.Context, c *Container, force, remo
 	}
 
 	if c.IsService() {
-		canStop, err := c.canStopServiceContainer()
-		if err != nil {
-			return err
+		var pods []string
+		for _, id := range c.state.Service.Pods {
+			if _, err := c.runtime.LookupPod(id); err != nil {
+				if errors.Is(err, define.ErrNoSuchPod) {
+					continue
+				}
+				return err
+			}
+			pods = append(pods, id)
```

Comment:
storing the pods is not necessary, you can just return the error here.

Reply:
A service container can be associated with multiple pods.  I prefer to give one complete list rather than erroring out multiple times in a row.

Reply:
but you do not use the pods var, you still use c.state.Service.Pods in the error message

Reply:
Oops.  With the latest changes, I will follow your initial suggestion. The list is now always up to date.